src/GoogleAdWordsEventForwarder.js

```diff
@@ -70,51 +71,151 @@
+            adWordEvent.google_conversion_label = conversionLabel;
+
+            if (mPEvent.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
+                && mPEvent.ProductAction.TransactionId) {
+                adWordEvent.google_conversion_order_id = mPEvent.ProductAction.TransactionId;
+            }
+
+            if (mPEvent.CurrencyCode) {
+                adWordEvent.google_conversion_currency = mPEvent.CurrencyCode;
+            }
 
+            if (mPEvent.ProductAction.TotalAmount) {
+                adWordEvent.google_conversion_value = mPEvent.ProductAction.TotalAmount;
+            }
 
+            adWordEvent.google_custom_params = getCustomProps(mPEvent, isPageEvent);
+            return adWordEvent;
+        }
+
+        // gtag Events
+        function generateGtagEvent(mPEvent, conversionLabel, isPageEvent) {
+            var conversionPayload = {
+                'send-to': gtagSiteId + '/' + conversionLabel
+            };
+
+            var customProps = getCustomProps(mPEvent, isPageEvent);
+
+            var payload = Object.assign({}, conversionPayload, customProps);
+
+            return payload;
+        }
+
+        function generateGtagCommerceEvent(mPEvent, conversionLabel, isPageEvent) {
+            var conversionPayload = {
+                'send-to': gtagSiteId + '/' + conversionLabel
+            };
+
+            var customProps = getCustomProps(mPEvent, isPageEvent);
+
+            if (mPEvent.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
+                && mPEvent.ProductAction.TransactionId) {
                 if (event.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
                     && event.ProductAction.TransactionId) {
                     adWordEvent.google_conversion_order_id = event.ProductAction.TransactionId;
                 }
+                conversionPayload.order_id = mPEvent.ProductAction.TransactionId;
+            }
 
-                if (event.CurrencyCode) {
-                    adWordEvent.google_conversion_currency = event.CurrencyCode;
-                }
+            if (mPEvent.CurrencyCode) {
+                conversionPayload.currency = mPEvent.CurrencyCode;
+            }
 
-                if (event.ProductAction.TotalAmount) {
-                    adWordEvent.google_conversion_value = event.ProductAction.TotalAmount;
-                }
+            if (mPEvent.ProductAction.TotalAmount) {
+                conversionPayload.value = mPEvent.ProductAction.TotalAmount;
+            }
+
+            var payload = Object.assign({}, conversionPayload, customProps);
+
+            return payload;
+        }
 
-                adWordEvent.google_custom_params = getCustomProps(event, isPageEvent);
+        // Sends final event to Google or queues if Google isn't ready
+        function sendOrQueueEvent(conversionPayload) {
+            if (window.gtag && forwarderSettings.enableGtag == 'True') {
```

Comment:
Putting this comment here, although it pertains to other lines of code, all referenced here as well:

in `logCommerce` and `logPageEvent`, we run `generateEvent` or `generateCommerceEvent`, and then `sendOrQueueEvent`. The issue is in `generateEvent` and `generateCommerceEvent`, we repeat lines 193-197 here.  In `generateEvent` and `generateCommerceEvent`, if neither `window.gtag` nor `window.google_trackConversion exist`, then eventPayload = false, and we never get to `sendOrQueueEvent` (see lines 210-216 and 227-233). 

This logic should to be refactored so that we `sendOrQueueEvent` if neither `gtag` nor `google_trackConversion` exist.